app/_components/types/OrderFormData.ts

```diff
@@ -16,12 +16,15 @@ export interface Address {
   streetAddress1: string,
   streetAddress2: string,
   townCity: string,
+  state: string,
   zip: string,
+  orders: number
 }
 
 export type Addresses = Address[];
 
-
+// Elliot: originally the recipAddress was just a number that had the index of the address from the Addresses type
+// I'm not sure if we need to save the address in the orderItem explicitly, so I'll just leave a comment about it to discuss later with Darryl
```

Comment:
My logic here was something like:
The interface (object) we are trying to represent is a single item of a customer's order. In other words, what information do they submit when they click "add to cart?"

If we only need the idx for displaying the data in a sorted manner for the client, I don't think we need it as a part of the OrderItem, which will be sent to the back end as an individual order to be printed and delivered along with the arrangement.

In short, I think it may make sense to instead define a CartItem interface that extends OrderItem and just includes the deliveryDateIdx as an additional field. We can then use that interface for all cart-related logic while keeping the interface clean as far as what we send to the business. 

We can move the sorting algorithm to a method in the CartContext, eg CartContext.addToCart() or CartContext.updateCart() - that would make the algorithm reusable across the website. 

So if we imagine a cart that looks like this:
`Cart {
deliveryDates: [date1, date2]
order: [[order1], [order2]]
}`
User selects another arrangement, selects date1 for the item, and clicks "add to cart" ->
-> delivery date is checked against existing delivery dates;
-> orderItem is added to order at the index that is found;
`Cart {
deliveryDates: [date1, date2],
cartItems: [[order1, order3], [order2]]
}
`

This is then consumed by the component with useCart() for rendering elements.

Does that sound okay? Let's chat more about it, maybe we can do some coworking next week